docs/rule/no-curly-component-invocation.md

```diff
@@ -1,115 +1,92 @@
-
-### Blacklisting components without dashes
-Since Ember 3.10 components have not required dashes in their name, e.g. `{{datepicker}}`. To help the linter throw an error on these curly component invocations, which it would otherwise have thought to be a helper or property, you can explicitly add it to the `disallow` section of the rule's config. Any curly statements matching an entry in `disallow` will throw a lint error.
-
-```js
-module.exports = {
-  rules: {
-    'no-curly-component-invocation': {
-        disallow: [
-        'heading',
-        'datepicker',
-      ],
-    },
-  },
-};
-```
-
-To get a list of of all the components in your app which don't have dashes in their name run the following code in your browser's Developer Tools Console when your app has loaded:
-
-``` js
-var componentsWithoutDashes = Object.keys(require.entries)
-  .filter(name => name.includes('/components/'))
-  .filter(name => !name.includes('-'))
-  .map(name => {
-    let path = name.split('/components/');
-    return path.pop();
-  })
-  .uniq();
-
-copy(JSON.stringify(componentsWithoutDashes));
-```
-
-### Matching on components without dashes in their name
-Before Ember 3.10 components were required to have at least one dash, `-`, in their name. By default this rule assumes that all components have `-` in their name. If you'd like to change this behaviour to match all curly invocations, even those without dashes then set the `requireDash` option to `false`.
-
-```js
-module.exports = {
-  rules: {
-    'no-curly-component-invocation': {
-        requireDash: false,
-    },
-  },
-};
-```
-
-### Related Rules
-
-* [no-args-paths](no-args-paths.md)
+## no-curly-component-invocation
+
+There are two ways to invoke a component in a template: curly compoment syntax
+(`{{my-component}}`), and angle bracket syntax (`<MyComponent />`). The
+difference between them is syntactical. You should favour angle bracket syntax
+as it improves readability of templates, i.e. disambiguates components from
+helpers, and is also the future direction Ember is going with the Octane
+Edition.
+
+This rule checks all the curly braces in your app and warns about those that
+look like they could be component invocations.
+
+### Examples
+
+- `{{foo}}` ✅
+  (simple mustache without `-` in the path is likely to be a property; unless
+  `noImplicitThis` is set or in `disallow` list and not a scoped variable)
+
+- `{{foo.bar}}` ✅
+  (simple mustache with nested path is likely to be a nested property; unless
+  `noImplicitThis` is set)
+
+- `{{foo-bar}}` ❌ (angle brackets: `<FooBar />`)
+  (simple mustache with `-` in the path is more likely a component than a
+  property or helper; unless in `allow` list)
+
+- `{{nested/component}}` ❌ (angle brackets: `<Nested::Component />`)
+  (simple mustache with `/` in the path is more likely a component than a
+  property or helper; unless in `allow` list)
+
+- `{{42}}` ✅
```

Comment:
I assume this covers `{{true}}`, `{{false}}`, `{{null}}` and `{{undefined}}` right?

Reply:
yes